### PR TITLE
Add fonts-rpm-macros to Mariner

### PR DESCRIPTION
**Pull request by 0xba1a**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Add new package fonts-rpm-macros to Mariner

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local build

fonts-rpm-macros provides the macros used in most of the font packages in Fedora. To import those font packages to Mariner with minimal effort, fonts-rpm-macros package is inevitable.

As it is an rpm-macros package, it should be available in the chroot while doing the SPEC parsing. So, this package should be present ideally in the SPECS directory. But as it calls for additional ~30 packages to be moved into SPECS, I'm keeping it in the SPECS-EXTENDED directory. Even the fedora documentation says that this package doesn't need to be part of the build environment by default. It should be added only while building font packages. So, keeping it in SPECS-EXTENDED and adding the package in pkggen_core file while building a font package will be a viable approach.

### CODE REVIEW

File: SPECS-EXTENDED/fonts-rpm-macros/fonts-rpm-macros.spec [hunk 1]
```diff
@@ -0,0 +1,240 @@
+Vendor:         Microsoft Corporation
```
Comment (mfrw) [inline SPECS-EXTENDED/fonts-rpm-macros/fonts-rpm-macros.spec:1]: Can we please attach a full build pipeline of both ARM/AMD with it.
Just to ensure we are not breaking anything :)

Comment (0xba1a) [reply SPECS-EXTENDED/fonts-rpm-macros/fonts-rpm-macros.spec:1]: Triggered a full build. BuildId: 372290

Comment (0xba1a) [reply SPECS-EXTENDED/fonts-rpm-macros/fonts-rpm-macros.spec:1]: AMD64 BuildRPMs complete. ARM64 BuildRPMs failed on doxigen, which is not relevant to this PR. Retriggered.

**Review by mfrw — APPROVED** [review]
LGTM :)

Lets take care of the Github PR checks :)

**Pull request merged**